core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -253,26 +251,8 @@ private void deliver() {
        * be in unprocessed.
        */
       boolean stalled = unprocessed.readableBytes() == 0;
-
-      if (endOfStream && stalled) {
-        boolean havePartialMessage = nextFrame != null && nextFrame.readableBytes() > 0;
```

Comment:
Added this back in, and added test cases in AbstractServerStreamTest and MessageDeframerTest. The deframer will notify the listener on close if it has a partial message - this is ignored by clients, since they don't set end-of-stream, but used by server to call deframeFailed when appropriate.